pkg/client/auth/kerberos/spnego_windows.go

```diff
@@ -21,18 +25,29 @@ func NewSPNEGOCredentials(serverUrl string, config ClientConfig) (credentials.Pe
 
 func (s *spnego) GetRequestMetadata(ctx context.Context, uri ...string) (map[string]string, error) {
 	// TODO: keep clientContext for multiple requests
+	log := ctxlogrus.Extract(ctx)
 
 	cred, err := negotiate.AcquireCurrentUserCredentials()
 	if err != nil {
 		return nil, err
 	}
-	defer cred.Release()
+	defer func() {
+		if err := cred.Release(); err != nil {
+			err = errors.WithStack(err)
```

Comment:
Do you need the `errors.WithStack(err)` if you are doing `logging.WithStacktrace`?  Seems redundant?

Reply:
`logging.WithStacktrace` prints the stack trace captured via `errors.WithStack`.